Survey/Sources/Presentation/Modules/Login/LoginViewModel.swift

```diff
@@ -21,6 +21,41 @@ extension LoginViewModel: ViewModel {
         let activityTracker = ActivityTracker(false)
         let output = Output()
 
+        let isEmailValid = Publishers.CombineLatest(input.$email, input.emailTrigger)
+            .map { $0.0.isEmailValid && !$0.0.isEmpty }
+
+        let isPasswordValid = Publishers.CombineLatest(input.$password, input.passwordTrigger)
+            .map { $0.0.isPasswordMustLeast8Letters && !$0.0.isEmpty }
+
+        isEmailValid
+            .assign(to: \.isEmailValid, on: output)
+            .store(in: &output.cancelBag)
+
+        isPasswordValid
+            .assign(to: \.isPasswordValid, on: output)
+            .store(in: &output.cancelBag)
+
+        Publishers.CombineLatest(isEmailValid, isPasswordValid)
+            .map { $0.0 && $0.1 }
+            .assign(to: \.isLoginEnabled, on: output)
+            .store(in: &output.cancelBag)
+
+        input.logInTrigger
+            .filter { output.isLoginEnabled }
+            .map { _ in
+                self.useCase.login(email: input.email, password: input.password)
+                    .trackError(errorTracker)
+                    .trackActivity(activityTracker)
+                    .asDriver()
+            }
+            .switchToLatest()
+            .sink(receiveValue: {
+                KeychainAccess.remove()
+                KeychainAccess.token = $0 as? KeychainToken
+                output.isLoggedInSuccessfully = KeychainAccess.isLoggedIn
```

Comment:
Should we create a use case to store Token instead of setting value directly like this and assign it to isLogInSuccessfully?
```suggestion 
                .map { _ in
                     self.logInUseCase.login(email: input.email, password: input.password)
                           .trackError(errorTracker)
                           .trackActivity(activityTracker)
                }
                .map { token in
                    self.storeTokenUseCase.store(token).asDriver()
                }
                .assign(to: \.isLoggedInSuccessfully, on: output)
                .store(in: &output.cancelBag)
```

Reply:
Good point 👍  I will move them to a use  case.

Reply:
Is that Ok if I store token in LoginUseCase?, for example:
```swift
    func login(email: String, password: String) -> Observable<Bool> {
        loginRepository
            .login(email: email, password: password)
            .map { token in
                KeychainAccess.remove()
                KeychainAccess.token = token as? KeychainToken
                return KeychainAccess.isLoggedIn
            }
            .eraseToAnyPublisher()
    }
```

Reply:
I don't think so. We used to meet this case in another project and we all agree to separate it to a use case so that you can re-use it in another case like refresh token.

Reply:
My suggestion is to create a use case for storing token and chain it in ViewModel after you login successfully. What do you think?

Reply:
> Separate it to a use case so that you can re-use it in another case like refresh token and to make sure each use case only do their own job

Yes you're right, it's a good point 👍 . I'm gonna fix it, thanks 😄 .

Reply:
@ducbm051291 I fixed at https://github.com/nkhanh44/ic-survey-ios/pull/15/commits/ee45f0aaed723fa827bab88295e61c0d6d5afaf5, please help me review this again, thanks. 🙇